### ISSUE TITLE
future update ?

### ISSUE THREAD

**Issue by rlacr0ix** [blesta/module-proxmox#35]
I am still interested in Blesta and this module, but there is a huge lack of functionality on this module, for example:

- CloudInit support for KVMs
- IPv6 support
- backup support

Do you have any update plans for this module ?

**Comment by TeYroXOfficial** [comment]
What exactly would you want to have?
I assume, CloudInit should have IP support and password reset?
About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.

And backup uhmm.. Maybe one day :D

**Comment by rlacr0ix** [comment]
In fact, CloudInit is a module that can be initialized on templates with a Cloud image (https://pve.proxmox.com/wiki/Cloud-Init_Support).

this allows, for example, a cloudinit template of an Ubuntu cloud image, to create a new virtual machine (a Clone of the template) and to assign it additional parameters like Password, Username (Generally we put root), SSH Key, configuration network (IPv4/IPv6) and dns configuration (dns server etc)

This allows you to deploy a virtual machine without installation and is what is used by hosts to deploy their client's servers automatically.

As for IPv6, it is actually similar to IPv4 (we simply give a single address)

And it would therefore be necessary that from Blesta, we then assign a list of IPs (v4 or v6) and a list of template (vmid) on each server group (To be able to manage all these resources within a cluster)

**Comment by Fedsd1** [comment]
> What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> 
> And backup uhmm.. Maybe one day :D

Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?

**Comment by TeYroXOfficial** [comment]
> > What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> > And backup uhmm.. Maybe one day :D
> 
> Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?

Wouldn't it be better to set it per package instead of allowing client to enable such feature?

**Comment by Fedsd1** [comment]
> > > What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> > > And backup uhmm.. Maybe one day :D
> > 
> > 
> > Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?
> 
> Wouldn't it be better to set it per package instead of allowing client to enable such feature?

This is also possible, may I ask if this plugin supports proxmox8.0.4, I set the swap to 0 in the plugin but it doesn't seem to take effect, and the generated lxc container still gives 512m swap

**Comment by rlacr0ix** [comment]
and for CloudInit ? I think it would be very interesting to set this up for KVM

**Comment by TeYroXOfficial** [comment]
> > > > What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> > > > And backup uhmm.. Maybe one day :D
> > > 
> > > 
> > > Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?
> > 
> > 
> > Wouldn't it be better to set it per package instead of allowing client to enable such feature?
> 
> This is also possible, may I ask if this plugin supports proxmox8.0.4, I set the swap to 0 in the plugin but it doesn't seem to take effect, and the generated lxc container still gives 512m swap

Hard to say, I didn't updated my proxmox to 8.x yet, so can't test it. Also what version of module are you using?

**Comment by TeYroXOfficial** [comment]
> and for CloudInit ? I think it would be very interesting to set this up for KVM

CloudInit or TAP/TUN? I just did basic IPv4 implementation for CI now. Working now on resetting passwords and few other things.

**Comment by rlacr0ix** [comment]
Proxmox provides CloudInit APIs, with this you can update all IP information (v4/v6) DNS passwords and SSH key, it only works for KVM, not LXC

**Comment by TeYroXOfficial** [comment]
> Proxmox provides CloudInit APIs, with this you can update all IP information (v4/v6) DNS passwords and SSH key, it only works for KVM, not LXC

Yes, I know that :) That's why I asked about CI or TAP/TUN. I though that this second thing is also required for KVM to be implemented but is seems like this one is only for LXC, so nevermind.

**Comment by rlacr0ix** [comment]
ooh im sorry, when I created this issue, it was mainly for KVM because I don't use LXC, sorry I should have been clearer

**Comment by Fedsd1** [comment]
> > > > > What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> > > > > And backup uhmm.. Maybe one day :D
> > > > 
> > > > 
> > > > Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?
> > > 
> > > 
> > > Wouldn't it be better to set it per package instead of allowing client to enable such feature?
> > 
> > 
> > This is also possible, may I ask if this plugin supports proxmox8.0.4, I set the swap to 0 in the plugin but it doesn't seem to take effect, and the generated lxc container still gives 512m swap
> 
> Hard to say, I didn't updated my proxmox to 8.x yet, so can't test it. Also what version of module are you using?

3.0.0

**Comment by TeYroXOfficial** [comment]
> > > > > > What exactly would you want to have? I assume, CloudInit should have IP support and password reset? About IPv6, I don't really know how to implement it. I could just do single IP's implementation like current v4 have.
> > > > > > And backup uhmm.. Maybe one day :D
> > > > > 
> > > > > 
> > > > > Can you provide a button for the LXC client's interface to turn on and off TUN/TAP?
> > > > 
> > > > 
> > > > Wouldn't it be better to set it per package instead of allowing client to enable such feature?
> > > 
> > > 
> > > This is also possible, may I ask if this plugin supports proxmox8.0.4, I set the swap to 0 in the plugin but it doesn't seem to take effect, and the generated lxc container still gives 512m swap
> > 
> > 
> > Hard to say, I didn't updated my proxmox to 8.x yet, so can't test it. Also what version of module are you using?
> 
> 3.0.0

Weird then. It should work perfectly fine, no changes in API was made for swap.
You ca write to me on discord @teyrox or on Blesta Discord so we will figure it out
